### ISSUE TITLE
R.trampoline

### ISSUE THREAD

**Issue by ccorcos** [ramda/ramda#1484]
Since JS doenst have tail-call recursion, it seems like R.trampoline would be a good addition to this library

http://raganwald.com/2013/03/28/trampolines-in-javascript.html

**Comment by CrossEye** [comment]
We did have one for a while, although used only for the LazyList extension, which was removed in #857.

Now that there are discussions underway about creating lazy structures again (#1482), I agree that we need to think about trampolining again.

Just a minor nit, though.  "Since JS doenst have tail-call recursion" is not at all accurate.  JS certainly does have tail recursion.  The problem is that until recently there was no specification that insisted on an optimization of tail calls, and that even now there are practically no implementations that do support it.  (As of this writing, according to [Kangax's table][kg], Babel gets 1/2, Webkit r191276 and Kinoma XS6 get 2/2.  Everyone else gets 0/2.)


  [kg]: https://kangax.github.io/compat-table/es6/#test-proper_tail_calls_(tail_call_optimisation)
